src/include/duckdb/common/vector_operations/generic_executor.hpp

```diff
@@ -67,23 +81,32 @@ struct StructTypeState {
 template <class A_TYPE>
 struct StructTypeUnary {
 	A_TYPE a_val;
+	bool is_null;
 
 	using STRUCT_STATE = StructTypeState<1>;
 
-	static bool ConstructType(STRUCT_STATE &state, idx_t i, StructTypeUnary<A_TYPE> &result) {
+	bool ContainsNull() {
+		return is_null;
+	}
+
+	static StructTypeUnary<A_TYPE> ConstructType(STRUCT_STATE &state, idx_t i) {
 		auto &a_data = state.child_data[0];
 		auto a_idx = a_data.sel->get_index(i);
-		if (!a_data.validity.RowIsValid(a_idx)) {
-			return false;
+		StructTypeUnary<A_TYPE> result;
+		result.is_null = !a_data.validity.RowIsValid(a_idx);
+		if (!result.ContainsNull()) {
+			auto a_ptr = UnifiedVectorFormat::GetData<A_TYPE>(a_data);
```

Comment:
It would be great if we could call `A_TYPE::ConstructType` here instead of directly using the `UnifiedVectorFormat`. That also means `a_data ` should be `A_TYPE::STRUCT_STATE`.

This is a bit more of a rework - but it makes all of these constructors more robust and allows nested types to work correctly within structs as well (including structs within structs).